x/pubsub/types/proposal.go

```diff
@@ -0,0 +1,184 @@
+package types
+
+import (
+	fmt "fmt"
+	"net/url"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	ProposalTypeAddPublisher              = "AddPublisher"
+	ProposalTypeRemovePublisher           = "RemovePublisher"
+	ProposalTypeAddDefaultSubscription    = "AddDefaultSubscription"
+	ProposalTypeRemoveDefaultSubscription = "RemoveDefaultSubscription"
+)
+
+var _ govtypes.Content = &AddPublisherProposal{}
+var _ govtypes.Content = &RemovePublisherProposal{}
+var _ govtypes.Content = &AddDefaultSubscriptionProposal{}
+var _ govtypes.Content = &RemoveDefaultSubscriptionProposal{}
+
+//////////////////////////
+// AddPublisherProposal //
+//////////////////////////
+
+func NewAddPublisherProposal(title string, description string, domain string, address string, proofUrl string, caCert string) *AddPublisherProposal {
+	return &AddPublisherProposal{
+		Title:       title,
+		Description: description,
+		Domain:      domain,
+		Address:     address,
+		ProofUrl:    proofUrl,
+		CaCert:      caCert,
```

Comment:
Nit: `Ca` -> `CA` for go style consistency? Or is ID a special case

Reply:
It's a habit of camel-casing I suppose, this is also true of e.g. `ProofUrl`, but the linter only seems to care about ID.